src/mlpack/methods/ann/layer/add.hpp

```diff
@@ -100,6 +100,9 @@ class Add
   //! Get the output size.
   size_t OutputSize() const { return outSize; }
 
+  //! Get the size of weights.
+  size_t const& WeightSize() { return outSize; }
```

Comment:
```suggestion
  const size_t WeightSize() const { return outSize; }
```
The function can be marked `const`; returning a `size_t const&` here will end up being ignored (the reference is superfluous; the compiler will choose to just pass by value and make a copy since `sizeof(size_t) == sizeof(size_t&)`; and our style guide prefers the `const` first. :smile:

Reply:
Actually I am not sure, but `const size_t` might throw a warning.  See what the compiler does and then you can choose whether to do `size_t WeightSize() const` or `const size_t WeightSize() const`. :)